### ISSUE TITLE
Ability to define linter ruleset in a file

### ISSUE THREAD

**Issue by timotheeguerin** [microsoft/typespec#3011]
This would allow us to have the spec repo define the set of rule they want to onboard/disable without being blocked by the azure core packages release as well as combining from multiple sources.

## Proposal

Allow to extends ruleset from a file. The file must be yaml and in the same structure as the ruleset structure today

```ts
interface RuleSet {
  extends?: string[]
  enable?: Record<string, boolean;
  disable?: Record<string, string>;
}
```

## Usage

To use a file instead of extending a ruleset defined in a package name we can point to the file with the `file:` prefix

```yaml
linter:
  extends: 
    - file:path/to/file.yaml
```

### Example file content
```yaml
extends: 
  - "@typespec/best-practices/recommended"
  - "@typespec/http/recommended"
enable: 
  "@typespec/best-practices/new-rule": true 
disable: 
  "@typespec/best-practices/foo": "This rule is too much"
```

### Example usage:

In tspconfig.yaml
```yaml
linter:
  extends: 
    - file:./../my-common-rules.yaml
```

### Example of what the azure-rest-api-specs repo could do 
```yaml
linter:
  extends: 
    - file:../../../typespec-rulesets/data-plane.yaml
    - file:../../../typespec-rulesets/arm.yaml # or
```

## Other uses
this pattern would also enable use to have a linter rule versioning system where we create a new ruleset file when we add new linter rule and yopu need to use the latest when you update

**Comment by markcowl** [comment]
est: 8

**Comment by markcowl** [comment]
Decided not to take this up for now, but see if there is interest in this in the community

**Comment by allenjzhang** [comment]
Closing in favor of https://github.com/Azure/typespec-azure/issues/654
